### PR TITLE
feat/docs: CL external incentives tool

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

This PR modifies the script to add the ability to create external incentives via gauges and wait for an epoch so that it is converted into an incentive record.

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

File: tests/cl-go-client/main.go
Comment (stackman27) [inline tests/cl-go-client/main.go:310]: can we also check gauge Records after we distribute? FilledEpoch should be +1 and distribtued Coins should be appended to the gauge struct

wait also i'm surprised this didn't fail here(https://github.com/osmosis-labs/osmosis/blob/036b94d3e220959e1ecc6050674c06de60f548f6/x/incentives/keeper/distribute.go#L393)

Comment (stackman27) [reply tests/cl-go-client/main.go:310]: also should we consider all the created incentives regardless of poolId <> gaugeId association and distribute tokens to them as well?

Comment (p0mvn) [reply tests/cl-go-client/main.go:310]: The purpose of this is to make sure that incentive records are created. This print gets the job done.

I don't think what you're requesting is helpful for this tool. However, it would be a good assertion to make in tests

File: tests/cl-go-client/main.go
Comment (mattverse) [inline tests/cl-go-client/main.go:270]:
```suggestion
	//Create gauge
```

File: tests/cl-go-client/main.go
Comment (mattverse) [inline tests/cl-go-client/main.go:267]: what do you mean by "if the duration is too long"? Are you referring to the epoch duration? This is the case where there aren't any epoch registered though right?

File: tests/cl-go-client/main.go
Comment (mattverse) [inline tests/cl-go-client/main.go:394]: out of curiosity, what's the reason we need to do this?

File: tests/cl-go-client/main.go
Comment (mattverse) [inline tests/cl-go-client/main.go:438]:
```suggestion
		log.Println("pool already exists. Tweak expectedPoolId variable if you want another pool, current expectedPoolId: ", expectedPoolId)
```

File: tests/cl-go-client/main.go
Comment (mattverse) [inline tests/cl-go-client/main.go:286]: does this get printed for an hour?

Comment (mattverse) [reply tests/cl-go-client/main.go:286]: maybe we can set a timer and retry every minute

**Review by mattverse — APPROVED** [review]
utACK.

A little concerned about the 

> log.Println("current epoch after gauge creation", currentEpochResponse.CurrentEpoch)
> 			log.Println("waiting for next epoch...")

getting logged for an hour, otherwise lgtm

File: tests/cl-go-client/main.go
Comment (p0mvn) [reply tests/cl-go-client/main.go:267]: Like if it is 1 full day.

The script is currently set up for localosmosis where epoch duration is 60 seconds.

File: tests/cl-go-client/main.go
Comment (p0mvn) [reply tests/cl-go-client/main.go:286]: On localosmosis, unintuitively, an "hour" epoch is actually 60 seconds so this shouldn't be a concern.

If this is used on testnet, there is no "hour" epoch so the script would fail early.

Thi script is mainly used for testing incentives on localosmosis. If there is an infinite loop somewhere, the user can just kill the script

Comment (mattverse) [reply tests/cl-go-client/main.go:286]: Ah I see, now I understand why everything was so confusing while reviewing this, now everything became more clearer

File: tests/cl-go-client/main.go
Comment (ValarDragon) [inline tests/cl-go-client/main.go:63]: Are we going to run into a problem trying to use this for cl-testnet?

Comment (p0mvn) [reply tests/cl-go-client/main.go:63]: The epoch identifier can manually be changed to "day" here.

A bigger problem is that we would have to wait for the epoch to pass for a while so that the incentive records are created

Comment (p0mvn) [reply tests/cl-go-client/main.go:63]: I will try running this against testnet before I merge and document all the changes that would be necessary

**Comment by p0mvn** [comment]
OK, I'm going to merge this script to unblock the external incentives PR branched off of this. I asked Ibrar to test this out to make sure that all needed functionality is present

**Pull request merged**
